config/gslib.cmake

```diff
@@ -0,0 +1,79 @@
+set(LIBP_GS_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/3rd_party/gslib/)
+set(LIBP_GS_LIB 3rd_party/gslib/libgs.a)
+set(OGS_SOURCE_DIR ${LIBP_GS_SOURCE_DIR}/ogs)
+
+# =============================================================
+# Build gslib
+# =============================================================
+
+# Copy source into CMake build dir.  We do this since gslib is built in-source,
+# and we want to keep source tree clean.
+FetchContent_Declare(
+    libp_gs_content
+    URL ${CMAKE_CURRENT_SOURCE_DIR}/3rd_party/gslib
+)
+FetchContent_GetProperties(libp_gs_content)
+if (NOT libp_gs_content)
+    FetchContent_Populate(libp_gs_content)
+endif()
+set(LIBP_GS_SOURCE_DIR ${libp_gs_content_SOURCE_DIR})
```

Comment:
There are a lot of name like LIBP_GS_SOURCE_DIR suggesting that GS is related to libP. That's something we should change.

Reply:
Since NekRS and Nek5000 use two different binaries of gslib, I needed to disambiguate the variables somehow.  In config/nek5000.cmake, there are already variables/targets related to Nek5000's gslib.  So here, I sort of "namespaced" the variables/targets for the gslib that came with libP.  

Would you like a different way to namespace these variables?  Maybe NEKRS_GS_SOURCE_DIR, etc?

Reply:
Can we "namespace" the nek5000 gslib instead?

Reply:
Sure, will do